### ISSUE TITLE
data recorder memory leak

### ISSUE THREAD

**Issue by yupinghuang** [ovro-lwa/lwa-issues#445]
```
Jul 25 14:29:08 lwacalim01 dr-vslow-1[1655763]: Exception in thread StorageLogger:
Jul 25 14:29:08 lwacalim01 dr-vslow-1[1655763]: Traceback (most recent call last):
Jul 25 14:29:08 lwacalim01 dr-vslow-1[1655763]:  File "/home/pipeline/.conda/envs/datarecorder/lib/python3.6/threading.py", line 916, in _bootstrap_inner
Jul 25 14:29:08 lwacalim01 dr-vslow-1[1655763]:    self.run()
Jul 25 14:29:08 lwacalim01 dr-vslow-1[1655763]:  File "/home/pipeline/.conda/envs/datarecorder/lib/python3.6/threading.py", line 864, in run
Jul 25 14:29:08 lwacalim01 dr-vslow-1[1655763]:    self._target(*self._args, **self._kwargs)
Jul 25 14:29:08 lwacalim01 dr-vslow-1[1655763]:  File "/home/pipeline/proj/lwa-shell/ovro_data_recorder/monitoring.py", line 342, in main
Jul 25 14:29:08 lwacalim01 dr-vslow-1[1655763]:    self._manage_quota()
Jul 25 14:29:08 lwacalim01 dr-vslow-1[1655763]:  File "/home/pipeline/proj/lwa-shell/ovro_data_recorder/monitoring.py", line 284, in _manage_quota
Jul 25 14:29:08 lwacalim01 dr-vslow-1[1655763]:    remove_process = Popen(['/bin/rm', '-rf'] + chunk, stdout=DEVNULL, stderr=DEVNULL)
Jul 25 14:29:08 lwacalim01 dr-vslow-1[1655763]:  File "/home/pipeline/.conda/envs/datarecorder/lib/python3.6/subprocess.py", line 729, in __init__
Jul 25 14:29:08 lwacalim01 dr-vslow-1[1655763]:    restore_signals, start_new_session)
Jul 25 14:29:08 lwacalim01 dr-vslow-1[1655763]:  File "/home/pipeline/.conda/envs/datarecorder/lib/python3.6/subprocess.py", line 1295, in _execute_child
Jul 25 14:29:08 lwacalim01 dr-vslow-1[1655763]:    restore_signals, start_new_session, preexec_fn)
Jul 25 14:29:08 lwacalim01 dr-vslow-1[1655763]: OSError: [Errno 12] Cannot allocate memory
```

**Comment by caseyjlaw** [comment]
Data recorders seem to be working now and disk space at the expected level. Was there a solution other than restarting the dr services?

**Comment by yupinghuang** [comment]
Sooo I can see the dr_visibilities's memory footprint growing from 2% last Fri to 9% today on calim02 onward.

I can see that Jayce is working on this already...?

**Comment by jaycedowell** [comment]
Yeah, I'm logging RSS size for dr-vslow-1 and 2 this morning to see if I can narrow down where this is coming from.

**Comment by jaycedowell** [comment]
I've stared at this for a while and I can't find the leak.  I've tried not recording, turning off different blocks within the pipeline, turning off the etcd updates, and tightening the memory usage inside the blocks but nothing seems to stop it.  It does seem to grow a little more slowly now for whatever that is worth.  It is strange that this only seems to happen with the slow vis recorder but maybe that is more related to slow being used continuously?
